health/visualizer/explorer/croissant_surfer_openml.ipynb

```diff
@@ -0,0 +1,348 @@
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/markdown": [
+       "Scrapped 5435 datasets for openml"
+      ],
+      "text/plain": [
+       "<IPython.core.display.Markdown object>"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "sns.set_style(\"white\")\n",
+    "num_rows = df.select(pl.len()).collect().item()\n",
+    "display(Markdown(f\"Scrapped {num_rows} datasets for {source}\"))\n",
+    "body = pl.col(\"body\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "id": "26bb4aa8-c1cf-4e9c-a314-33ea55cc97c8",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = np.array(df.collect())"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "id": "0893bb88-e1b1-4242-b663-b0d05ca53dc4",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "(5435, 15)\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(a.shape)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "id": "ce091182-e69f-4ab1-884f-1da8344e2430",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def extract_first_description_and_all_urls_per_entry(json_strings):\n",
+    "    descriptions = []\n",
+    "    all_urls = []  # This will be a list of lists\n",
+    "    description_pattern = re.compile(r'\"description\": \"(.*?)\"(?=[,}])', re.DOTALL)\n",
+    "    url_pattern = re.compile(r'\"url\": \"(.*?)\"(?=[,}])', re.DOTALL)\n",
+    "    \n",
+    "    for string in json_strings:\n",
+    "        # Search for the first description\n",
+    "        description_match = description_pattern.search(string)\n",
+    "        if description_match:\n",
+    "            description = description_match.group(1).encode('utf-8').decode('unicode_escape')\n",
+    "            descriptions.append(description)\n",
+    "        else:\n",
+    "            descriptions.append('No description found')\n",
+    "        \n",
+    "        # Search for URLs and keep them in a list corresponding to the current string\n",
+    "        url_matches = url_pattern.findall(string)\n",
+    "        if url_matches:\n",
+    "            urls = [match.encode('utf-8').decode('unicode_escape') for match in url_matches]\n",
+    "            all_urls.append(urls)  # Append list of URLs for the current entry\n",
+    "        else:\n",
+    "            all_urls.append(['No URL found'])  # Append a placeholder if no URLs are found\n",
+    "    \n",
+    "    return descriptions, all_urls"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "id": "57460c3b-a655-4ebb-a43e-22f82868e446",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "descriptions, urls = extract_first_description_and_all_urls_per_entry([str(x[0]) for x in a]) #get a list of croissant descriptions and urls from the openml crawl dump"
```

Comment:
nit: You can use [`json.loads`](https://docs.python.org/3/library/json.html#json.loads) to deserialize JSON and avoid regex pattern matching. In the `crawler`, we should probably store them in JSON in the parquet files in the first place though

Reply:
done